### ISSUE TITLE
Scenario logging level is not respected when Platform Event Storage Logging Level is null

### ISSUE THREAD

**Issue by arbokrad** [jongpie/NebulaLogger#538]
### Package Edition of Nebula Logger

Unlocked Package

### Package Version of Nebula Logger

v4.11

### New Bug Summary

**Setup**
Org wide settings:
- LoggingLevel = `ERROR`
- Enabled = true
- Enable Saving = true
- Platform Event Storage Location = CUSTOM_OBJECTS
- Platform Event Storage Logging Level = (not set - i think this matters later)

Scenario rule settings for SCENARIO_A:
- User Logging Level =  `INFO`
- Enabled = true
- Enable Platform Event Storage Override = (not set)
- Platform Event Storage Location = (not set)
- Enable Log Retention Policy = true
- Days to Retain Logs = 90

**Code**
```java
Logger.setScenario('SCENARIO_A');
Logger.info('any message');
Logger.saveLog();
```

**Expected results**
- A single log and log entry generated, belonging to `SCENARIO_A`

**Actual results**
- Logger finds our rules for `SCENARIO_A`
- A `LogEntryEvent__c` is created (visible from the streaming console)
- Event handler is reached, and runs the `filterLogEntryEventsToSave` method
- Event is filtered out
- No log/log entry is created

When I change the org wide settings like so:
- Platform Event Storage Logging Level = `INFO`

... and re-run the same code snippet, the results are:
- A single log and log entry generated, belonging to `SCENARIO_A`

It seems like a bug in how it picks logging level to keep or filter out in `filterLogEntryEventsToSave` where scenario log level is not respected, but only when the Platform Event Storage Logging Level is null?

**Comment by jongpie** [comment]
A quick update on this - I'm hoping to finally fix this as part of the Winter '25 release (`v4.15.0`).

I've heard from several teams (including a few internal teams) in the last couple of months that this is impacting them. And since this issue causes data loss, with no indication that anything is wrong (since there's no error - it's an issue in the logic itself), it's not a great bug to have sitting out there for this long.

My hope is to get this fixed first in the unlocked package ASAP, which will help get feedback to ensure that everything is working before the fix is released for the managed package later this year in `v4.15.0`.

**Comment by jongpie** [comment]
Welp.... after writing my last comment, I thought about the issue a bit more, did some tinkering, and I was able to fix the bug 🥳 I've created PR #722 for this. And since I'm _still_ wrapping up the Summer '24 release (`v4.14.0` - PR #700), this bugfix will be included in the next managed package release, which should be finally released next week (I hope).

**Comment by jongpie** [comment]
This has been fixed (I hope 😅 ) in [release `v4.13.16`](https://github.com/jongpie/NebulaLogger/releases/tag/v4.13.16)
